scripts/document.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+if [ -z `which jazzy` ]; then
```

Comment:
In both cases `jazzy` is installed/is not istalled `which jazzy` returns non-empty string, doesn't it?
In general I'm not sure `which` can detect installed gems correctly.

Should we try to use
```sh
gem list -i "jazzy"
```
to check if `jazzy` is installed?

Reply:
Agree, we shouldn't rely on output text. And that's the same reason `gem list` also doesn't suite.
WDYT about `hash jazzy 2>/dev/null` check? It will give the correct return code and won't produce any output (to stdout).

Reply:
Same concern for `hash`. I'm not sure it can detect all installed gems.

On the other hand `gem list -I`/`gem list -I` specifically checks for installed/no-installed gem and it has straightforward return codes. Can we try 
```sh
if gem list -I "jazzy"; then
...
fi
```
?

Reply:
The intention here is to check whether there's an executable `jazzy` and not whether there's a gem `jazzy`. `hash` will search in `$PATH` to make sure command is executable, whereas `gem list` will check for installed gem which may not be executable.
In this case `gem list` will work since `jazzy` is installed as executable, but this check doesn't satisfy the intention of this check.

Reply:
But in order to install it we stick to `gem` as far as I understand. This is why I proposed to use gem's infrastructure.

`hash ...` is ok too.

Reply:
Yes, if there's no executable `jazzy` which we're using later, we're trying to install it through `gem` because... it's a gem :)

In this particular case, result should be the same. Do you prefer `gem list` way? I can change it.

Reply:
`hash` is ok.

My point is if it is gem then let's use specified parameter for `gem list` that does the job (checking if gem is installed).